blog/content/post/2024-03-15-uefi-sb-implementation-mpl.md

```diff
@@ -0,0 +1,385 @@
+
+   ```bash
+   sudo apt-get install git virtualenv python3-pip
+   ```
+
+- Clone the OFSV repository and update linked submodules
+
+  ```bash
+  git clone https://github.com/Dasharo/open-source-firmware-validation
+  cd open-source-firmware-validation
+  git submodule update --init --checkout
+  ```
+
+- Prepare virtualenv
+
+  ```bash
+  python3 -m virtualenv venv
+  source venv/bin/activate
+  pip install -r requirements.txt
+  ```
+
+If you want, you can run the tests in QEMU. To set it up, follow these steps:
+
+- Navigate to `scripts/ci` in the OSFV repository.
+- Create a `qemu-data` directory and create a disk `qcow` file there.
+
+  ```bash
+  mkdir scripts/ci
+  qemu-img create -f qcow2 qemu-data/hdd.qcow 20G
+  ```
+
+- Download the Ubuntu installer and run QEMU with the `os-installer` argument
+
+  ```bash
+  INSTALLER_PATH=path/to/installer ./qemu-run.sh graphic os-installer
+  ```
+
+- Go through the installation and close QEMU.
+
+QEMU is now ready for OSFV. You run it by executing the following command:
+
+```bash
+./qemu-run.sh graphic os-installer
+```
+
+### Test implementation
+
+Development of test cases in the Robot Framework consists of defining
+reusable keywords and utilizing them in concrete scenarios. Dasharo OSFV
+introduces many such keywords, which allow reading and writing to the terminal,
+navigating menus, etc. This allows the developer to implement test cases with
+much more ease. Below is an example of a test case, which verifies that Secure
+Boot does not allow booting files that are not signed.
+
+```robot
+SBO004.001 Attempt to boot file without the key from Shell (firmware)
+    [Documentation]    This test verifies that Secure Boot blocks booting a file
+    ...    without a key.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO004.001 not supported
+    Mount ISO As USB    ${CURDIR}/../scripts/secure-boot/images/NOT_SIGNED.img
+    # 1. Make sure that SB is enabled
+    Power On
+    ${sb_menu}=    Enter Secure Boot Menu And Return Construction
+    Enable Secure Boot    ${sb_menu}
+    # Save Changes And Reset
+    # Changes to Secure Boot menu takes action immediately, so we can just reset
+    Reset System
+    Boot Efi File Should Fail    hello.efi    NOT_SIGNED
+```
+
+As you can see, the test case is brief and easily readable thanks to its use of
+predefined keywords.
+
+The test suite can be found in the
+[OSFV repository](https://github.com/Dasharo/open-source-firmware-validation/blob/main/dasharo-security/secure-boot.robot)
+and the detailed description of each test is available on the
+[Unified Test Documentation](https://docs.dasharo.com/unified-test-documentation/dasharo-security/206-secure-boot)
+page.
+
+The test suite can be run using the following command:
+
+```bash
+robot -L TRACE \
+  -v ansible_config:yes \
+  -v rte_ip:<rte_ip> \
+  -v snipeit:no \
+  -v config:mpl-pip4 \
+  -v device_ip:<device_ip> \
+  dasharo-security/secure-boot.robot
+```
```

Comment:
Please explain what all those parameters mean, many of those are Dasharo/Zarhus Certification Lab-related and specific to OSFV infrastructure. Graphics which is in OSFV repo could help understanding what is what.

Reply:
For osfv-specific flags I linked the repo's readme
1aae9529980da9d1019c722a5af6ac3c47a5c0d2